sensor/kubernetes/enforcer/cronjob/suspend.go

```diff
@@ -0,0 +1,62 @@
+package cronjob
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/branding"
+	kubernetesPkg "github.com/stackrox/rox/pkg/kubernetes"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/sensor/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	batchV1      = "batch/v1"
+	batchBeta1v1 = "batch/beta1v1"
+)
+
+// Suspend suspends the cron job
+func Suspend(ctx context.Context, client kubernetes.Interface, deploymentInfo *central.DeploymentEnforcement) (err error) {
+	depName := deploymentInfo.GetDeploymentName()
+	depType := deploymentInfo.GetDeploymentType()
+	forcePatch := true
+
+	if ok, apiErr := utils.HasAPI(client, batchV1, kubernetesPkg.CronJob); ok && apiErr == nil {
```

Comment:
Can you think of a way to not query this API each time a CronJob needs to be suspended? This may not be a lot of overhead, and we may drop the support for `batch/v1beta1` altogether in 4.0.0 release, but would be nice to reduce the usage if it does not lead to way more complicated code.

Reply:
I'm going to leave it in there if especially `batch/v1beta1` support is going to be drooped.